pkg/authn/keychain.go

```diff
@@ -97,6 +117,34 @@ func (dk *defaultKeychain) Resolve(target Resource) (Authenticator, error) {
 		}
 	} else {
 		f, err := os.Open(filepath.Join(os.Getenv("XDG_RUNTIME_DIR"), "containers/auth.json"))
+		if err != nil {
+			return nil, nil
+		}
+		defer f.Close()
+		cf, err = config.LoadFromReader(f)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return cf, nil
+}
+
+func (dk *defaultKeychain) Resolve(target Resource) (Authenticator, error) {
+	dk.mu.Lock()
+	defer dk.mu.Unlock()
+
+	var cf *configfile.ConfigFile
+	if dk.configFilePath == "" {
```

Comment:
Should we sync.Once this so we don't lock the _entire_ `Resolve`? Why was the mutex added originally?